src/main/java/esi/backend/service/RequestService.java

```diff
@@ -42,4 +52,19 @@ public void updateRequest(Request request) {
     public void deleteRequest(UUID id) {
         requestRepository.deleteById(id);
     }
+
+    public ResponseEntity<List<Request>> getAllRequestsByCustomerId(UserDetails currentUser, long customerId){
+        Optional<Customer> optionalCustomer = customerRepository.findById(customerId);
+        if (optionalCustomer.isEmpty()){
+            return new ResponseEntity<>(HttpStatus.NOT_FOUND);
+            //throw new ResourceNotFoundException("Customer with id" + customerId + "not found");
+        }
+        Customer customer = optionalCustomer.get();
+        if (!currentUser.getUsername().equals(customer.getUsername())){
+            return new ResponseEntity<>(HttpStatus.UNAUTHORIZED);
+        }
```

Comment:
I (personal preference, tho) would get the Customer type on the first line already to make the code shorter and a bit more clear:
```
Customer customer = customerRepository.findById(customerId).orElse(null);
if (customer == null) {
  return new ResponseEntity<>(HttpStatus.NOT_FOUND);
}
if (!currentUser.getUsername().equals(customer.getUsername())) {
  return new ResponseEntity<>(HttpStatus.UNAUTHORIZED);
}
```